### PR TITLE
🧳 Move zen generation script and fix tests

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

It makes more sense to have it in scripts since the dataset is only used for testing purposes.

I realised that each time you run the generation script for zen, it shuffles the dataset, resulting in some tests to fail.
This PR fixes it.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: tests/test_sft_trainer.py [hunk 1]
```diff
@@ -1172,7 +1172,7 @@ def test_sft_trainer_eval_packing(self):
             )
 
             self.assertEqual(len(trainer.train_dataset["input_ids"]), 46)  # w/ this dataset, we end up with 46 seqs
-            self.assertEqual(len(trainer.eval_dataset["input_ids"]), 5)  # w/ this dataset, we end up with 5 seqs
+            self.assertEqual(len(trainer.eval_dataset["input_ids"]), 6)  # w/ this dataset, we end up with 5 seqs
```
Comment (kashif) [inline tests/test_sft_trainer.py:1175]:
```suggestion
            self.assertEqual(len(trainer.eval_dataset["input_ids"]), 6)  # w/ this dataset, we end up with 6 seqs
```

**Review by kashif — APPROVED** [review]
